File: 코드리뷰/이세영(0112leesy)/3일차/코드리뷰_마라톤.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class main {
    static int[][] arr;
    static int dist(int i, int j) {
        return Math.abs(arr[j][0]-arr[i][0]) + Math.abs(arr[j][1]-arr[i][1]);
    }

    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int N = Integer.parseInt(br.readLine());
        arr = new int[N][2];
        for(int i=0; i<N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            arr[i][0] = Integer.parseInt(st.nextToken());
            arr[i][1] = Integer.parseInt(st.nextToken());
        }
        int total = 0;
        for(int i=0; i<N-1; i++) {
            total += dist(i,i+1);
        }
        int min = total;
        for(int i=1; i<N-1; i++) {
            int skip = dist(i-1, i) + dist(i, i+1) - dist(i-1, i+1);
            int skipped = total - skip;
            if(skipped < min) min = skipped;
        }
        System.out.println(min);
    }

}

File: 코드리뷰/장문주(jmj2359)/3일차/코드리뷰_마라톤.py
#i점 건너뛸때, 건너뛰지 않을 때 거리 차이
def dist(d,i):
    d1 = abs(d[i][0]-d[i-1][0])+abs(d[i][1]-d[i-1][1])+abs(d[i+1][0]-d[i][0])+abs(d[i+1][1]-d[i][1])
    d2 = abs(d[i+1][0]-d[i-1][0])+abs(d[i+1][1]-d[i-1][1])
    return d1-d2
n = int(input())
d = []

for i in range(n):
    d += [list(map(int,input().split()))]



c = 0
for i in range(1,n-1):
    if (c==0) or (dist(d,i) >= c):
        c = dist(d,i)
t = 0
for i in range(1,n):
    t+=abs(d[i][0]-d[i-1][0])+abs(d[i][1]-d[i-1][1])

print(t-c)
